services/s3.cfc

```diff
@@ -290,6 +290,172 @@ component {
         return apiResponse;
     }
 
+
+    /**
+     * This action initiates a multipart upload and returns an upload ID. This upload ID is used to associate all of the parts in the specific multipart upload. You specify this upload ID in each of your subsequent upload part requests (see UploadPart). You also include this upload ID in the final request to either complete or abort the multipart upload request.
+     * https://docs.aws.amazon.com/AmazonS3/latest/API/API_CreateMultipartUpload.html
+     * 
+     *
+     * @Bucket: The name of the bucket to which to initiate the upload
+     * @Key: Object key for which the multipart upload is to be initiated.
+     */
+    public any function createMultipartUpload(
+        required string Bucket,
+        required string Key
+    ){
+
+         var requestSettings = api.resolveRequestSettings( argumentCollection = arguments );
+
+        var queryParams = { 'uploads': '' };
+        for (
+            var key in arguments
+        ) {
+            // if ( len( arguments[ key ] ) ) queryParams[ utils.parseKey( key ) ] = arguments[ key ];
+        }
+        var apiResponse = apiCall(
+            requestSettings,
+            'POST',
+            '/#arguments.Key#',
+            queryParams,
+            {
+                // "x-amz-acl": "public-read"
+            }
+        );
+        if ( apiResponse.statusCode == 200 ) {
+            apiResponse[ 'data' ] = utils.parseXmlDocument( apiResponse.rawData );
+        }
+        return apiResponse;
+        
+    }
+
+    public any function putPart(
+        required string Bucket,
+        required string Key,
+        required string UploadId,
+        required numeric PartNumber, 
+        required any body
+
+    ){
+        var requestSettings = api.resolveRequestSettings( argumentCollection = arguments );
+        dump("putPart");
+        var queryParams = { 
+            'partNumber': arguments.partNumber, 
+            "uploadId": arguments.uploadId 
+        };
+
+        for (
+            var key in arguments
+        ) {
+            // if ( len( arguments[ key ] ) ) queryParams[ utils.parseKey( key ) ] = arguments[ key ];
+        }
+        var apiResponse = apiCall(
+            requestSettings,
+            'PUT',
+            '/#arguments.Key#',
+            queryParams,
+            {
+                // "x-amz-acl": "public-read"
+            },
+            arguments.body
+        );
+
+        if ( apiResponse.statusCode == 200 ) {
+            // apiResponse[ 'data' ] = utils.parseXmlDocument( apiResponse.rawData );
```

Comment:
I don't know what the api response looks like here, is it an XML document? If not, can this 200 check be removed?

Reply:
Yeah, it should be XML for the most part.

Reply:
Actually in this case it just returns headers rather than any body content so I had to remove the XML Parsing